service/tablet_allocator.cc

```diff
@@ -1049,10 +1092,39 @@ class load_balancer {
                     return r2_hosts.contains(dst.host);
                 };
 
-                lblogger.debug("Replica sets of tablets being co-located: ({}: {}), ({}, {})",
-                        global_tablet_id{table, t1.tid}, r1, global_tablet_id{table, t2.tid}, r2);
+                lblogger.debug("Replica sets of tablets being co-located: ({}: {}), ({}, {}), rack_awareness={}",
+                        global_tablet_id{table, t1.tid}, r1, global_tablet_id{table, t2.tid}, r2, rack_awareness);
+
+                if (rack_awareness) {
+                    auto r1_sets = per_rack_replica_sets(r1);
+                    auto r2_sets = per_rack_replica_sets(r2);
+
+                    for (auto&& [rack, r1_s] : r1_sets) {
+                        auto it = r2_sets.find(rack);
+                        if (it == r2_sets.end()) {
+                            // When adding a new rack, e.g. from RF-1 to RF, only one of sibling tables might have
+                            // presence on all racks, so we can wait for migrations to settle before we proceed
```

Comment:
Which migrations are you referring to?

Reply:
Let's say we're increasing from 2 to 3 racks, and RF 3. Tablet 0 might still be at rack 1 and rack 2 (since previously there were only 2 racks). And tablet 1 already at all 3 racks. So this is a transient state I expect the balancer to correct, by spreading the tablet load on all racks available. Do we have trigger logic for this?

Reply:
> Let's say we're increasing from 2 to 3 racks, and RF 3. Tablet 0 might still be at rack 1 and rack 2 (since previously there were only 2 racks). And tablet 1 already at all 3 racks. So this is a transient state I expect the balancer to correct, by spreading the tablet load on all racks available. Do we have trigger logic for this?

No, we don't.

Reply:
But it should be fine if we enforce RF=#racks. So increasing RF from to 2 to 3 means we're allocating new replica from the new rack. Rack aware colocation requires equal distribution of replicas across racks. Perhaps merge can aid in this scenario by migrating a replica of a tablet that doesn't use all racks available (when rack awareness is turned on), so merge doesn't get stuck waiting for a condition that might never be satisfied by the balancer.

Reply:
Although this is something that should possibly be handled by balancer. Detecting replica set of a tablet is not fairly spread across all racks available and emitting migrations for correcting that. What do you think?

Reply:
But we can start by falling back to rack-unaware merging in this case, it doesn't make things worse.

Reply:
I think fallback works fine now, but for us to enable table with views, which requires consistent base-view pairing, we'll really need the changes in balancer to unblock merge. Right? Anyway, it can be postponed until we really need it